.github/workflows/deploy-docs.yml

```diff
@@ -9,13 +9,20 @@ on:
     types:
       - published
 
+permissions:
+  # Give the GITHUB_TOKEN write permission to open a PR with the changes to the switcher.json file.
+  contents: write
+  pull-requests: write
+
 jobs:
   run:
     runs-on: ubuntu-latest
+
     steps:
     - uses: actions/checkout@v4
       with:
         fetch-depth: 0
+        ref: main
```

Comment:
@Conengmo and @merschformann do you need this? That means any change in the code won't be tested in the docs until it is on `main`.

Reply:
This is here to make sure that docs are only ever released from the _main_ branch, thus, reflecting docs in the state the _main_ branch is in. A push to _main_ is also a trigger of this workflow.
It does not feel fully necessary, if the events would change to only be pushes to _main_ and `release` (this one could technically result in none-_main_ being released if release is defined on branch). I am not sure what `pull_request` is for. It is just explicit about limiting docs releases to be coming from _main_.
(just my two cents)

Maybe the triggers of the workflow and the checkout should be revised? :thinking: 

cc @Conengmo